src/test/scala/nl/knaw/dans/easy/stage/EasyStageDatasetSpec.scala

```diff
@@ -16,156 +16,144 @@
 
     // Note that files.xml specifies NONE for data/path/to/file.txt
 
-    createFileAndFolderSdos(dataDir, DATASET_SDO, OPEN_ACCESS_FOR_REGISTERED_USERS) shouldBe a[Success[_]]
-    readFileToString(fileMetadataFile) should include ("<visibleTo>ANONYMOUS</visibleTo>")
-    readFileToString(fileMetadataFile) should include ("<accessibleTo>NONE</accessibleTo>")
-    deleteDirectory(sdoSetDir)
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, OPEN_ACCESS_FOR_REGISTERED_USERS) shouldBe a[Success[_]]
+    readFileToString(fileMetadataFile.toFile) should include ("<visibleTo>ANONYMOUS</visibleTo>")
+    readFileToString(fileMetadataFile.toFile) should include ("<accessibleTo>NONE</accessibleTo>")
 
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Success[_]]
-    readFileToString(fileMetadataFile) should include ("<visibleTo>ANONYMOUS</visibleTo>")
-    readFileToString(fileMetadataFile) should include ("<accessibleTo>NONE</accessibleTo>")
-    deleteDirectory(sdoSetDir)
-
-    tmpProps.delete()
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Success[_]]
+    readFileToString(fileMetadataFile.toFile) should include ("<visibleTo>ANONYMOUS</visibleTo>")
+    readFileToString(fileMetadataFile.toFile) should include ("<accessibleTo>NONE</accessibleTo>")
   }
 
-  "run" should "create SDO sets from test bags (proof the puddings by eating them with easy-ingest)" in {
-    assume(canConnect(xsds))
-
-    createProps()
-
-    // license-by-url seems to require mocking web-access, probably beyond the purpose of this test
-    def useTestBag(f: File) = f.getName != "additional-license-by-url"
-
-    val testBags = new File ("src/test/resources/dataset-bags").listFiles().filter(useTestBag)
-    val emptyDataDir = new File("src/test/resources/dataset-bags/minimal/data")
-    val puddingsDir = new File ("target/sdoPuddings")
-    emptyDataDir.mkdir()
-
-    // clean up old results
-    FileUtils.deleteDirectory(puddingsDir)
+  "checkFilesInBag" should "result in Success if files argument is empty" in {
+    checkFilesInBag(Set.empty, testBagMedium) shouldBe a[Success[_]]
+  }
 
-    for (bag <- testBags) {
-      val sdoSetDir = new File(puddingsDir,bag.getName)
-      implicit val settings = createSettings(bag, sdoSetDir)
+  it should "result in Failure if passed list of one file that is not in the bag" in {
+    val result = checkFilesInBag(Set(Paths.get("data/NOT-README.md")), testBagMedium)
 
-      EasyStageDataset.run(settings) shouldBe a[Success[_]]
-      new File(sdoSetDir, "dataset/EMD").exists() shouldBe true
-      new File(sdoSetDir, "dataset/AMD").exists() shouldBe true
-      new File(sdoSetDir, "dataset/cfg.json").exists() shouldBe true
-      new File(sdoSetDir, "dataset/fo.xml").exists() shouldBe true
-      new File(sdoSetDir, "dataset/PRSQL").exists() shouldBe true
+    result shouldBe a[Failure[_]]
```

Comment:
I think this error message is not very clear when your try to match on a `Failure` when it isn't one.